lib/sync/remote-resin-io-device.coffee

```diff
@@ -128,20 +131,39 @@ exports.sync = ({ uuid, baseDir, destination, before, after, ignore, port = 22,
 	throw new Error("'destination' is a required sync option") if not destination?
 	throw new Error("'uuid' is a required sync option") if not uuid?
 
+	# Resolves with object with required device info or is rejected if API was not accessible. Resolved object:
+	#
+	#	{
+	#		fullUuid: <string, full resin.io device UUID>
+	#		isAUFS: <true|false, depending on if the device is running on AUFS>
+	#	}
+	#
 	getDeviceInfo = (uuid) ->
+		RequiredDeviceObjectFields = [ 'uuid', 'os_version' ]
+
+		# Returns a promise that is resolved with the API-fetched device object or rejected on error or missing requirement
+		ensureDeviceRequirements = (device) ->
+			# Ensure user is the owner of the device. This is also checked on the backend side.
+			resin.auth.getUserId()
+			.then (userId) ->
+				if userId isnt device.user.__id
+					throw new Error('Resin sync is permitted to the device owner only. The device owner is the user who provisioned it.')
+			.then -> # Ensure minimum required HostOS version for resin sync
+				ensureHostOSCompatibility(device.os_version, MIN_HOSTOS_RSYNC)
+			.then ->
+				throw new Error("Fetched device info is missing required field #{field}") for field in RequiredDeviceObjectFields when field not of device
+				return device
+
 		console.info("Getting information for device: #{uuid}")
 
 		resin.models.device.isOnline(uuid).then (isOnline) ->
 			throw new Error('Device is not online') if not isOnline
 			resin.models.device.get(uuid)
-		.tap (device) ->
-			# Ensure user is the owner of the device. This is also checked on the backend side.
-			resin.auth.getUserId().then (userId) ->
-				if userId isnt device.user.__id
-					throw new Error('Resin sync is permitted to the device owner only. The device owner is the user who provisioned it.')
-		.tap (device) ->
-			ensureHostOSCompatibility(device.os_version, MIN_HOSTOS_RSYNC)
-		.get('uuid') # get full uuid
+		.then(ensureDeviceRequirements)
+		.then ({ uuid, os_version }) ->
+			isAUFSPromise = ensureHostOSCompatibility(os_version, MIN_AUFS_RESINOS_VER)
```

Comment:
I'd also rather if we could detect the filesystem on the actual device itself, rather than having hardcoded semver versions (which has issues with custom os builds that have versions that are different to the main resin os versions)